### ISSUE TITLE
Nested properties and prop-types

### ISSUE THREAD

**Issue by mathieumg** [yannickcr/eslint-plugin-react#105]
Is it an oversight or a limitation that for the `react/prop-types` rule, no warning is emitted for missing nested properties.

For example, with:

```js
propTypes: {
  a: React.PropTypes.shape({
    b: React.PropTypes.shape({
      c: React.PropTypes.func
    })
  })
}
```

I get no warning when I use `this.props.a.b.d`. 

Thank you!

**Comment by Cellule** [comment]
If this is something we want to add support for. I fiddled with it a bit tonight and I got something working for quite a few nested cases.
It checks for `shape` `arrayOf` `objectOf` `oneOfType` and `instanceOf`.
I need to do some additional testing, but so far it look pretty solid. I can event do checks like
`this.props.a[0][someVar].e` and give warning about `e` if the propTypes are
```js
propTypes = {
  a: React.PropTypes.arrayOf(
    React.PropTypes.objectOf(
      React.PropTypes.shape({
        b: React.PropTypes.string
      })
    )
  )
};
```

Let me know if you want me to polish it.

**Comment by yannickcr** [comment]
Wow. That's definitely something we're interested in. I did not dive in your code yet but it seems great so far.

You can polish it, I will be very happy to merge it :)
